### ISSUE TITLE
form-control-static does not take into account form-group-sm and lg

### ISSUE THREAD

**Issue by Lukino2000** [twbs/bootstrap#15536]
hi everyone, like i wrote in title form-control-static does not take into account form-group sizes and padding top is incorrect on sm/lg size

**Comment by CoryMcDonald** [comment]
Do you have a JS Fiddle for an example?

**Comment by cvrebert** [comment]
Confirmed: http://jsbin.com/rodixo/1

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/rodixo/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by CoryMcDonald** [comment]
@cvrebert 
Do you think that the left-alignment be the same as the inputs?
http://jsbin.com/bopajehosi

**Comment by twbs-lmvtfy** [comment]
Hi @CoryMcDonald!

You appear to have posted a live example (http://jsbin.com/bopajehosi/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by peterblazejewicz** [comment]
@cvrebert 
Is this OK as fix?
```diff
diff --git a/less/forms.less b/less/forms.less
index bf15cc8..ca89051 100644
--- a/less/forms.less
+++ b/less/forms.less
@@ -308,14 +308,16 @@ input[type="checkbox"] {
 // Build on `.form-control` with modifier classes to decrease or increase the
 // height and font-size of form controls.
 //
-// The `.form-group-* form-control` variations are sadly duplicated to avoid the
-// issue documented in https://github.com/twbs/bootstrap/issues/15074.
+// The `.form-group-* form-control`, `.form-group-* form-control-static` 
+// variations are sadly duplicated to avoid the issue documented
+// in https://github.com/twbs/bootstrap/issues/15074.
 
 .input-sm {
   .input-size(@input-height-small; @padding-small-vertical; @padding-small-horizontal; @font-size-small; @line-height-small; @input-border-radius-small);
 }
 .form-group-sm {
-  .form-control {
+  .form-control,
+  .form-control-static {
     .input-size(@input-height-small; @padding-small-vertical; @padding-small-horizontal; @font-size-small; @line-height-small; @input-border-radius-small);
   }
 }
@@ -324,7 +326,8 @@ input[type="checkbox"] {
   .input-size(@input-height-large; @padding-large-vertical; @padding-large-horizontal; @font-size-large; @line-height-large; @input-border-radius-large);
 }
 .form-group-lg {
-  .form-control {
+  .form-control,
+  .form-control-static {
     .input-size(@input-height-large; @padding-large-vertical; @padding-large-horizontal; @font-size-large; @line-height-large; @input-border-radius-large);
   }
 }
```
dist:
```diff
diff --git a/dist/css/bootstrap.css b/dist/css/bootstrap.css
index 8dd893e..fb26f13 100644
--- a/dist/css/bootstrap.css
+++ b/dist/css/bootstrap.css
@@ -2691,19 +2691,23 @@ textarea.input-sm,
 select[multiple].input-sm {
   height: auto;
 }
-.form-group-sm .form-control {
+.form-group-sm .form-control,
+.form-group-sm .form-control-static {
   height: 30px;
   padding: 5px 10px;
   font-size: 12px;
   line-height: 1.5;
   border-radius: 3px;
 }
-select.form-group-sm .form-control {
+select.form-group-sm .form-control,
+select.form-group-sm .form-control-static {
   height: 30px;
   line-height: 30px;
 }
 textarea.form-group-sm .form-control,
-select[multiple].form-group-sm .form-control {
+textarea.form-group-sm .form-control-static,
+select[multiple].form-group-sm .form-control,
+select[multiple].form-group-sm .form-control-static {
   height: auto;
 }
 .input-lg {
@@ -2721,19 +2725,23 @@ textarea.input-lg,
 select[multiple].input-lg {
   height: auto;
 }
-.form-group-lg .form-control {
+.form-group-lg .form-control,
+.form-group-lg .form-control-static {
   height: 46px;
   padding: 10px 16px;
   font-size: 18px;
   line-height: 1.33;
   border-radius: 6px;
 }
-select.form-group-lg .form-control {
+select.form-group-lg .form-control,
+select.form-group-lg .form-control-static {
   height: 46px;
   line-height: 46px;
 }
 textarea.form-group-lg .form-control,
-select[multiple].form-group-lg .form-control {
+textarea.form-group-lg .form-control-static,
+select[multiple].form-group-lg .form-control,
+select[multiple].form-group-lg .form-control-static {
   height: auto;
 }
 .has-feedback {
```
http://jsbin.com/retuhi/2/edit?html,css,output
Can do PR if OK.

**Comment by twbs-lmvtfy** [comment]
Hi @peterblazejewicz!

You appear to have posted a live example (http://jsbin.com/retuhi/2/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by peterblazejewicz** [comment]
:twbs-lmvtfy: fixed

**Comment by cvrebert** [comment]
@peterblazejewicz Might as well go for it.

**Comment by philipstratford** [comment]
This still seems to be an issue in 3.3.2. In a div with .form-group-sm, a p element with .form-control-static still isn't aligned with its label. This fiddle demonstrates it - note the CSS I've added which is fixing the problem. http://www.bootply.com/bj3aNMHzR7

**Comment by mdo** [comment]
Ah, the problem is we don't set the `.control-label` size to match and don't also account for the `border` that's present on standard `.form-control`s. Another fix is at #16176.
